addons/mail/models/mail_thread.py

```diff
@@ -4533,6 +4533,13 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
         empty_messages = message.sudo()._filter_empty()
         empty_messages._cleanup_side_records()
         empty_messages.write({'pinned_at': None})
+        # Notify front-end of messages deletion for accurate counters
```

Comment:
Could remove comment, the message is semantically empty so it makes sense to notify the frontend for message deletion.

The current use of `mail.message/delete` is an actual `mail.message/unlink`, which happens actually rarely. IIRC this was a bug fix from deleting messages in technical settings.